### DISCUSSION TITLE
render sidemenucontroller when editor editable field is set to false.

### DISCUSSION THREAD

**Discussion by sijirama** [TypeCellOS/BlockNote#2499] [Ideas & Enhancements]
**Is your feature request related to a problem? Please describe.**
basically i'll like to be able to use the sidemenu controller when the editor editable field is set to false.

**Describe the solution you'd like**
to be able to use the sidemenu controller when editable field is false.

**Describe alternatives you've considered**
i genuinely do not know, maybe another field passed to the editor called, showSideMenu, that defaults to false.

**Additional context**
![Screenshot from 2024-04-11 23-57-54](https://github.com/TypeCellOS/BlockNote/assets/106167719/c8b459fb-32e4-4b45-a0b2-b622c341efe5)

**Comment by sijirama** [comment]
~~i can attempt this~~ ,  never mind.

i don't know how you would like this to be set, if you even want this to be set, you pass in sideMenu and editable already, would you like another option to be passed to handle this?

**Comment by matthewlipski** [comment]
Ah, the reason you're running into this issue is it's not the controller that figures out when and for which block the side menu should be shown. That's done by the [side menu plugin](https://github.com/TypeCellOS/BlockNote/blob/main/packages/core/src/extensions/SideMenu/SideMenuPlugin.ts), and the controller uses information from it to show, hide, and position the side menu in the UI.

So unfortunately, this isn't possible atm but it may be a good idea to let people configure which UI elements can remain active when the editor is read-only. Thoughts on this @YousefED?
